src/server/qgsrequesthandler.cpp

```diff
@@ -189,70 +189,77 @@ void QgsRequestHandler::setupParameters()
   {
-    QString inputString( mRequest.data() );
-
-    QDomDocument doc;
-    QString errorMsg;
-    int line = -1;
-    int column = -1;
-    if ( !doc.setContent( inputString, true, &errorMsg, &line, &column ) )
+    if ( mRequest.header( QStringLiteral( "Content-Type" ) ).contains( QStringLiteral( "json" ) ) )
     {
-      // XXX Output error but continue processing request ?
-      QgsMessageLog::logMessage( QStringLiteral( "Warning: error parsing post data as XML: at line %1, column %2: %3. Assuming urlencoded query string sent in the post body." )
-                                 .arg( line ).arg( column ).arg( errorMsg ) );
-
-      // Process input string as a simple query text
-
-      typedef QPair<QString, QString> pair_t;
-      QUrlQuery query( inputString );
-      QList<pair_t> items = query.queryItems();
-      for ( pair_t pair : items )
-      {
-        // QUrl::fromPercentEncoding doesn't replace '+' with space
-        const QString key = QUrl::fromPercentEncoding( pair.first.replace( '+', ' ' ).toUtf8() );
-        const QString value = QUrl::fromPercentEncoding( pair.second.replace( '+', ' ' ).toUtf8() );
-        mRequest.setParameter( key.toUpper(), value );
-      }
       setupParameters();
     }
     else
     {
-      // we have an XML document
-
-      setupParameters();
-
-      QDomElement docElem = doc.documentElement();
-      // the document element tag name is the request
-      mRequest.setParameter( QStringLiteral( "REQUEST" ), docElem.tagName() );
-      // loop through the attributes which are the parameters
-      // excepting the attributes started by xmlns or xsi
-      QDomNamedNodeMap map = docElem.attributes();
-      for ( int i = 0 ; i < map.length() ; ++i )
+      QString inputString( mRequest.data() );
+      QDomDocument doc;
+      QString errorMsg;
+      int line = -1;
+      int column = -1;
+      if ( !doc.setContent( inputString, true, &errorMsg, &line, &column ) )
       {
-        if ( map.item( i ).isNull() )
-          continue;
-
-        const QDomNode attrNode = map.item( i );
-        const QDomAttr attr = attrNode.toAttr();
-        if ( attr.isNull() )
-          continue;
-
-        const QString attrName = attr.name();
-        if ( attrName.startsWith( "xmlns" ) || attrName.startsWith( "xsi:" ) )
-          continue;
-
-        mRequest.setParameter( attrName.toUpper(), attr.value() );
+        // XXX Output error but continue processing request ?
+        QgsMessageLog::logMessage( QStringLiteral( "Warning: error parsing post data as XML: at line %1, column %2: %3. Assuming urlencoded query string sent in the post body." )
+                                   .arg( line ).arg( column ).arg( errorMsg ) );
+
+        // Process input string as a simple query text
+
+        typedef QPair<QString, QString> pair_t;
+        QUrlQuery query( inputString );
+        QList<pair_t> items = query.queryItems();
+        for ( pair_t pair : items )
```

Comment:
Do you think we could use a const reference for `pair`?

Reply:
I just moved that code (it was already there), but I'll see if it builds with `const`.